app/javascript/react/utils/thresholdEventHandlers.ts

```diff
@@ -0,0 +1,141 @@
+import { useState } from "react";
+import { debounce } from "lodash";
+import { useTranslation } from "react-i18next";
+
+import { updateAdjacentThresholds } from "./tresholdsUpdateAdjacent";
+
+interface Thresholds {
+  min: number;
+  low: number;
+  middle: number;
+  high: number;
+  max: number;
+}
+
+export const useThresholdHandlers = (
+  setThresholdValues: React.Dispatch<React.SetStateAction<Thresholds>>,
+  setInputValue: React.Dispatch<React.SetStateAction<string>>,
+  setActiveInput: React.Dispatch<React.SetStateAction<keyof Thresholds | null>>,
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>,
+  thresholdValues: Thresholds,
+  sliderRef: React.RefObject<HTMLDivElement>,
+  activeInput: keyof Thresholds | null,
+  inputValue: string
+) => {
+  const [inputDebounceTime] = useState(300);
+  const { t } = useTranslation();
+
+  const handleInputChange = (thresholdKey: keyof Thresholds, value: string) => {
+    const trimmedValue = value.trim();
+    setInputValue(trimmedValue);
+
+    if (trimmedValue === "") {
+      setErrorMessage(t("thresholdConfigurator.emptyInputMessage"));
+      return;
+    }
+
+    setInputValue(trimmedValue);
```

Comment:
For me at least, instead of setting InputValue twice, it is clearer to set it up only when the error appears and assign to input an empty value (I hope I understood the intention :())

```
 const trimmedValue = value.trim();

    if (trimmedValue === "") {
      setInputValue("");
      setErrorMessage(t("thresholdConfigurator.emptyInputMessage"));
      return;
    }

    setInputValue(trimmedValue);
```